### PR TITLE
Calendar Panel

### PR DESCRIPTION
**Pull request by Tommatheussen**
This is the frontend implementation of the [Calendar Panel PR](https://github.com/home-assistant/home-assistant/pull/10192)  in the main repo.
Since I could not find a proper Polymer Calendar library to use, I have set up the system to show each calendar in a card, listing the events. Users will be able to see all their events in the components (past, current and future), example below.

image

As you can see, there are greyed out events, which are the ones that have passed already. Current running events are green, future events are plain black text.

The calendar view does not support updating when the component itself updates yet, will likely still come. Refreshing the view does load new events.

### CODE REVIEW

File: panels/calendar/ha-panel-calendar.html [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            </div>
+          </template>
+        </paper-card>
+      </div>
+    </app-header-layout>
+  </template>
+</dom-module>
+
+<script>
+Polymer({
```
Comment (balloob) [inline panels/calendar/ha-panel-calendar.html]: Use ES6 classes

File: panels/calendar/ha-panel-calendar.html [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    },
+
+    _events: {
+      type: Object,
+    },
+  },
+
+  attached: function () {
+    this.hassChanged = this.hassChanged.bind(this);
+    this.computePlatforms().then(function(platforms) {
```
Comment (balloob) [inline panels/calendar/ha-panel-calendar.html]: Don't use bind, use arrow functions instead.

File: panels/calendar/ha-panel-calendar.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      this._platforms.forEach(function(platform, index) {
+        this.computeEvents(platform.name).then(function(events) {
+          platform.events = events;
+          this.notifyPath('_platforms.' + index + '.events');
+        }.bind(this));
+       }.bind(this));
+    }
+  },
+
+  computePlatforms: function () {
```
Comment (balloob) [inline panels/calendar/ha-panel-calendar.html]: Name is incorrect. This is not computing, this is fetching.

File: src/components/ha-cards.html [hunk 4]
```diff
@@ -107,6 +107,7 @@
     sensor: 5,
     binary_sensor: 6,
     mailbox: 7,
+    calendar: 8,
```
Comment (balloob) [inline src/components/ha-cards.html]: This is not needed, there are no calendar cards added with this PR

**Comment by balloob** [comment]
Out of scope for this PR, but we should really consider using http://intljusticemission.github.io/react-big-calendar/examples/index.html

File: panels/calendar/ha-panel-calendar.html [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+<link rel='import' href='../../src/components/ha-menu-button.html'>
+<link rel='import' href='../../src/resources/ha-style.html'>
+
+
+<dom-module id='ha-panel-calendar'>
+  <template>
+    <style include='ha-style'>
+      .event-active .row {
+        color: green;
```
Comment (balloob) [inline panels/calendar/ha-panel-calendar.html]: Don't hardcode colors, use our variables instead.

**Comment by c727** [comment]
Main PR closed

**Pull request closed without merging**